pkg/api/resource/quantity_proto.go

```diff
@@ -17,62 +17,287 @@ limitations under the License.
 package resource
 
 import (
-	"math/big"
+	"fmt"
+	"io"
 
-	inf "gopkg.in/inf.v0"
+	"github.com/gogo/protobuf/proto"
 )
 
-// QuantityProto is a struct that is equivalent to Quantity, but intended for
-// protobuf marshalling/unmarshalling. It is generated into a serialization
-// that matches Quantity. Do not use in Go structs.
-//
-// +protobuf=true
-type QuantityProto struct {
-	// The format of the quantity
-	Format Format `protobuf:"bytes,1,opt,name=format,casttype=Format"`
-	// The scale dimension of the value
-	Scale int32 `protobuf:"varint,2,opt,name=scale"`
-	// Bigint is serialized as a raw bytes array
-	Bigint []byte `protobuf:"bytes,3,opt,name=bigint"`
-}
+var _ proto.Sizer = &Quantity{}
 
-// ProtoTime returns the Time as a new ProtoTime value.
-func (q *Quantity) QuantityProto() *QuantityProto {
-	if q == nil {
-		return &QuantityProto{}
+func (m *Quantity) Marshal() (data []byte, err error) {
```

Comment:
I believe the godoc in the IDL covers it, but if not I will add.

On May 16, 2016, at 7:35 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/resource/quantity_proto.go
<https://github.com/kubernetes/kubernetes/pull/25243#discussion_r63444003>:

>
> -// ProtoTime returns the Time as a new ProtoTime value.
> -func (q *Quantity) QuantityProto() *QuantityProto {
> -	if q == nil {
> -		return &QuantityProto{}
> +func (m *Quantity) Marshal() (data []byte, err error) {

Document the format of the returned data-- I take it it's just a string in
the same format as the json output?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/25243/files/2a61025932c283f19dcc4e003cc623c57cea71e0#r63444003>